config/environments/production.rb

```diff
@@ -22,13 +22,13 @@
 
   # Disable serving static files from the `/public` folder by default since
   # Apache or NGINX already handles this.
-  config.public_file_server.enabled = true
+  config.public_file_server.enabled = ENV["RAILS_SERVE_STATIC_FILES"].present?
```

Comment:
Heroku sets this ENV variable and it's what this is set to in a new Rails 7.0 app.